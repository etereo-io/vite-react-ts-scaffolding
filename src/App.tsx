import "./assets/styles/global.css";

import "@fontsource/roboto/300.css";
import "@fontsource/roboto/400.css";
import "@fontsource/roboto/500.css";
import "@fontsource/roboto/700.css";

import { AppRoutes } from "@/app/components/AppRoutes";

import { AppProviders } from "./app/components/AppProviders";

export function App() {
  return (
    <AppProviders>
      <AppRoutes />
    </AppProviders>
  );
}
