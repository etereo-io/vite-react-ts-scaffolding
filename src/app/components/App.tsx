import "@/assets/styles/global.css";

import "../modules/modules";

import { AppRoutes } from "@/app/components/AppRoutes";

import { AppProviders } from "./AppProviders";

export function App() {
  return (
    <AppProviders>
      <AppRoutes />
    </AppProviders>
  );
}
