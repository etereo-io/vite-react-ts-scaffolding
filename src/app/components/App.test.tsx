import { render, screen } from "@testing-library/react";

import { App } from "./App";

describe("App", () => {
  test("should render login by default", () => {
    const { container } = render(<App />);

    expect(screen.getByTestId("login-page")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
