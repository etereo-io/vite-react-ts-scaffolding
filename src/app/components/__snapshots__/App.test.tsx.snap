// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render login by default 1`] = `
<div>
  <div
    class="min-h-screen flex"
    data-testid="login-page"
  >
    <div
      class="hidden lg:block lg:w-1/2 bg-cover bg-center bg-no-repeat"
      style="background-image: url("https://source.unsplash.com/random?wallpapers");"
    />
    <div
      class="w-full lg:w-1/2 flex items-center justify-center p-8 bg-white shadow-lg"
    >
      <div
        class="max-w-md w-full space-y-6"
      >
        <div
          class="text-center"
        >
          <div
            class="mx-auto w-12 h-12 bg-blue-600 rounded-full flex items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-lock w-6 h-6 text-white"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="11"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="11"
              />
              <path
                d="M7 11V7a5 5 0 0 1 10 0v4"
              />
            </svg>
          </div>
          <h1
            class="mt-4 text-3xl font-bold text-gray-900"
          >
            Sign in
          </h1>
        </div>
        <form
          class="space-y-4"
          novalidate=""
        >
          <div>
            <label
              class="block text-sm font-medium text-gray-700 mb-1"
              for="email"
            >
              Email Address
            </label>
            <input
              autocomplete="email"
              class="w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-blue-500"
              id="email"
              name="email"
              required=""
              type="email"
            />
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700 mb-1"
              for="password"
            >
              Password
            </label>
            <input
              autocomplete="current-password"
              class="w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-blue-500"
              id="password"
              name="password"
              required=""
              type="password"
            />
          </div>
          <div
            class="flex items-center"
          >
            <input
              class="w-4 h-4 text-blue-600 focus:ring-blue-500 border-gray-300 rounded"
              id="remember-me"
              name="remember-me"
              type="checkbox"
            />
            <label
              class="ml-2 block text-sm text-gray-700"
              for="remember-me"
            >
              Remember me
            </label>
          </div>
          <button
            class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 transition-colors"
            type="submit"
          >
            Sign In
          </button>
          <div
            class="flex items-center justify-between text-sm"
          >
            <a
              class="text-blue-600 hover:text-blue-500 transition-colors"
              href="#"
            >
              Forgot password?
            </a>
            <a
              class="text-blue-600 hover:text-blue-500 transition-colors"
              href="#"
            >
              Don't have an account? Sign Up
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
